{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ref\n",
      "Seq('AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAG...TGG', SingleLetterAlphabet())\n",
      "1330\n"
     ]
    }
   ],
   "source": [
    "for i in SeqIO.parse(\"reference.fa\", \"fasta\"):\n",
    "    print seq_record.id\n",
    "    print repr(seq_record.seq)\n",
    "    print len(seq_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCCTAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAGCGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AACGGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACAGAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTATT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCGCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAATTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAGTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTAACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTGTGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATTCTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACTCAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAAAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGCACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGGTGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AACCAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CATTAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTAATAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCCAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTCTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATGTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGAAAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGACTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATTTAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCGTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCCCGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAGCCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGATC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATGCGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGAACT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGAAAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGCCGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGATGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GACCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAGTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATGTAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTCGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAATAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AATACT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCCTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCGCCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GAACCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCTCTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTTTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAAAAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGCGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTCTGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCAGCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTGAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTGGTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTTTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAAATG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGCCGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTCTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCAATA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCTGGTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CATTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CAGGTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTCAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGGTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GAACTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTGCGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CACTAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGTGGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCGGCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CACCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAATGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTTCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACGGCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCGCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCTAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGGCCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTAAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCGGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTGGAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGTTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCGCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGGAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACACTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTCGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGGGCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATTTGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AACGTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTACTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAGTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACATTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGATTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACACTGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTCCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGAGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAAATT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCGGTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCACTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCGTCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCTCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTGCCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGTAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GATGCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATCGAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCGATA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGCGGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCGAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGACTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCGCCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTCTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACGGCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGTCCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTGAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGTACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTTAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAATCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCCACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGAGTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTTTCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATTACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAGGAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGAACT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGCAGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTAAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCGGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGACAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TACTCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGTGGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGAGCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGTCCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CAATAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAAAAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTTGTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGTTGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACGTCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTGCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTCTCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AACATC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTAAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCTGCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGCAGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGGCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGTCTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTGCGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTCAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGCGCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGAAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACTCGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCGGCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCATG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCTAAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TATTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAGTGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCGAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGCCCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCAGGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AACAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTAATG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTGAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGGCGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ATTTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CAAAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GAAAAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCGATC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCTGAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCGTCGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGACAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CATTAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCCTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GAGGTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACATCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CACCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGCACC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAAAAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CACCAG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCTTTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCCGGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACCAGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTATTA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GAATGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCAGGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTCCAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTCGTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GGGTATA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AAAGCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACTGTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTGTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTCCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCTTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AGTCCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GTATCT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAATCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACGCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TGGTGT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CATGCG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCCGA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CTGGTC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTCAT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCTAAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GCGGGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TTTTTT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGTTAC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCTGAA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "AACATT\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGCACCA\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TAATGG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "TCGCCC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GATCTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "ACTCGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CCCGGC\n",
      "<class 'Bio.Seq.Seq'>\n",
      "GAACTG\n",
      "<class 'Bio.Seq.Seq'>\n",
      "CGGCAG\n",
      "<class 'Bio.Seq.Seq'>\n"
     ]
    }
   ],
   "source": [
    "for i in SeqIO.parse(\"reads.fa\", \"fasta\"):\n",
    "    print i.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import AlignIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Sequences must all be the same length",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-18-77181a20a294>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0malignment\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mAlignIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"reads.fa\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"fasta\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/Users/David/anaconda/lib/python2.7/site-packages/Bio/AlignIO/__init__.pyc\u001b[0m in \u001b[0;36mread\u001b[0;34m(handle, format, seq_count, alphabet)\u001b[0m\n\u001b[1;32m    425\u001b[0m     \u001b[0miterator\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mformat\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mseq_count\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malphabet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    426\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 427\u001b[0;31m         \u001b[0mfirst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0miterator\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    428\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    429\u001b[0m         \u001b[0mfirst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/David/anaconda/lib/python2.7/site-packages/Bio/AlignIO/__init__.pyc\u001b[0m in \u001b[0;36mparse\u001b[0;34m(handle, format, seq_count, alphabet)\u001b[0m\n\u001b[1;32m    372\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    373\u001b[0m         \u001b[0;31m# This imposes some overhead... wait until we drop Python 2.4 to fix it\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 374\u001b[0;31m         \u001b[0;32mfor\u001b[0m \u001b[0ma\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    375\u001b[0m             \u001b[0;32myield\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    376\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/David/anaconda/lib/python2.7/site-packages/Bio/AlignIO/__init__.pyc\u001b[0m in \u001b[0;36m_SeqIO_to_alignment_iterator\u001b[0;34m(handle, format, alphabet, seq_count)\u001b[0m\n\u001b[1;32m    274\u001b[0m         \u001b[0mrecords\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mSeqIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mformat\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malphabet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    275\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 276\u001b[0;31m             \u001b[0;32myield\u001b[0m \u001b[0mMultipleSeqAlignment\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecords\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malphabet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    277\u001b[0m     \u001b[0;32mraise\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    278\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/David/anaconda/lib/python2.7/site-packages/Bio/Align/__init__.pyc\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, records, alphabet, annotations)\u001b[0m\n\u001b[1;32m    174\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_records\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    175\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 176\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecords\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    177\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0malphabet\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    178\u001b[0m                 \u001b[0;31m# No alphabet was given, take a consensus alphabet\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/David/anaconda/lib/python2.7/site-packages/Bio/Align/__init__.pyc\u001b[0m in \u001b[0;36mextend\u001b[0;34m(self, records)\u001b[0m\n\u001b[1;32m    244\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    245\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mrec\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 246\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_append\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrec\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexpected_length\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    247\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    248\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/Users/David/anaconda/lib/python2.7/site-packages/Bio/Align/__init__.pyc\u001b[0m in \u001b[0;36m_append\u001b[0;34m(self, record, expected_length)\u001b[0m\n\u001b[1;32m    307\u001b[0m             \u001b[0;31m# raise ValueError(\"New sequence is not of length %i\" \\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    308\u001b[0m             \u001b[0;31m#                 % self.get_alignment_length())\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 309\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Sequences must all be the same length\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    310\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    311\u001b[0m         \u001b[0;31m# Using not self.alphabet.contains(record.seq.alphabet) needs fixing\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Sequences must all be the same length"
     ]
    }
   ],
   "source": [
    "alignment = AlignIO.read(\"reads.fa\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    ""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2.0
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}